### ISSUE TITLE
How can I help with adding Docker support?

### ISSUE THREAD

**Issue by gabrielruiu** [astrocorp42/rocket#1]
Hy,

I would be interested in supporting the project with developing additional features, more specifically, adding support for deploying to Docker. 

Can you provide a bit of feedback what you would consider for adding Docker support? How would the process look like? What config can the user add?

Any details would be helpful.

Thanks!

**Comment by z0mbie42** [comment]
Hi, It's a pleasure to read that!

The problem I see with `docker` is it use a client/server architecture, thus we cannot embed it easily.

Thus I think the first iteration of the `docker` provider should only be a wrapper of the `docker` CLI
(`exec.Command("sh", "-c", "docker....")`).

To implement a provider the first step is to provide a spec/usage examples on how you would like to use it in the `.rocket.toml` configuration file and the required fields (here an example for `github_releases` https://astrocorp.net/rocket/github_releases/#fields).

For examples should the `docker` provider be used like ?
```toml
# .rocket.toml
[docker]
docker_username = "$DOCKER_USERNAME"
docker_password = "$DOCKER_PASSWORD"
# images to push
images = [
  "astrocorp/rocket:lastest",
  "my-custom-registry/org/image:my-tag",
  "my-custom-registry/org/image:$VERSION", # can we use env vars here ? to spec
]
```

So I first invite you to list all the required fields, their types, and their default values.

The second step is to implement it in the following folder: https://github.com/astrocorp42/rocket/tree/master/providers

the third step is to add it to the `rocket` command: https://github.com/astrocorp42/rocket/blob/master/commands/rocket.go

The last step is to add tests :)

**Comment by z0mbie42** [comment]
I've started to spec the `docker` provider: https://astrocorp.net/rocket/docker/

I think it's not yet great because you cannot login to multiple docker registries.

Waiting for reviews/ideas

**Comment by z0mbie42** [comment]
implemented in v1.1.0 1623970c756888ffb8948627c136045041ead9c5

**Comment by gabrielruiu** [comment]
Hi, I see you already added the Docker support in the meantime :+1: 

I'm going on vacation for about 2weeks and when I come back I'll check to see if there's anything left in the TODO list and hopefully I can help out with that.
